gufe/settings/models.py

```diff
@@ -0,0 +1,118 @@
+"""
+Pydantic models used for storing settings.
+"""
+
+from datetime import date
+from typing import Literal, Optional, Union
+
+from openff.models.models import DefaultModel
+from openff.models.types import FloatQuantity
+from pydantic import Extra, FilePath, PositiveFloat
+from typing_extensions import TypedDict
+
+
+class SettingsBaseModel(DefaultModel):
+    """
+    Settings and modifications we want for all settings classes.
+    """
+
+    class Config:
+        extra = Extra.forbid
+        # Immutability in python is never strict.
+        # If developers are determined/stupid
+        # they can always modify a so-called "immutable" object.
+        # https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability
+        allow_mutation = False
+        arbitrary_types_allowed = False
+        smart_union = True
+
+
+class vdWScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class ElectrostaticScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class VdWSettings(SettingsBaseModel):
+    """
+    Settings for van der Waals force
+    """
+
+    combining_rules: Literal["Lorentz-Berthelot"]
```

Comment:
Can we add `pydantic` `Field` usage to each of these attributes, with a one-liner for what each one is?